### PR TITLE
create screenshot dir if plugin is used (closes #4477)

### LINKED ISSUE #4477: Investigate screenshot problems introduced in v1.6.0
- [x] https://github.com/DevExpress/testcafe-browser-provider-saucelabs/issues/53
 - [x] https://github.com/DevExpress/testcafe/issues/4440
 - [ ] https://github.com/DevExpress/testcafe/issues/4475

### PR DESCRIPTION
**Pull request by AlexKamaev**
I reverted the fix of https://github.com/DevExpress/testcafe/issues/4303
Since it leaded to failure in custom browser provers
Now the dir will be created for all plugins but not for local browser.
For local browser the dir is created by `browser-tools`

### CODE REVIEW

File: src/browser/provider/index.ts [hunk 1]
```diff
@@ -317,8 +319,11 @@ export default class BrowserProvider {
             else
                 await this._takeLocalBrowserScreenshot(browserId, screenshotPath);
```
Comment (AndreyBelym) [inline src/browser/provider/index.ts]: Looks like we won't create a directory in the code path where `_takeLocalBrowserScreenshot` is called.

Comment (AlexKamaev) [reply src/browser/provider/index.ts]: `testcafe-browser-tools` will
https://github.com/DevExpress/testcafe-browser-tools/blob/775f1012c454558a0fd58af1f576036e9b22c453/src/api/screenshot.js#L17

Comment (AndreyBelym) [reply src/browser/provider/index.ts]: It's very unobvious. At least, we should add a comment about it. 

But I would prefer to refactor the code to get both cases covered by `makeDir`.

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Pull request merged**